Libraries/LibIPC/Connection.h

```diff
@@ -81,38 +94,55 @@ class Connection : public ConnectionBase {
     {
     }
 
-    template<typename MessageType>
-    OwnPtr<MessageType> wait_for_specific_message()
+    ~Connection() override = default;
+
+    template<typename RequestType, typename... Args>
+    NonnullOwnPtr<typename RequestType::ResponseType> send_sync(Args&&... args)
     {
-        return wait_for_specific_endpoint_message<MessageType, LocalEndpoint>();
+        auto result = send<RequestType, Args...>(forward<Args>(args)...)->await();
```

Comment:
this implementation of send_sync is behaviorally very different from the prior implementation. This implementation will add an event loop spin to the stack, and continue processing other events. `ConnectionBase::wait_for_specific_endpoint_message_impl` on master completely blocks execution by checking only the one specific socket in question, and causing any other events or IPC messages to be queued until the correct response type is received.

Reply:
Yes, and for deprecated XMLHttpRequest's (the sync api version) we will need to maintain the old "actually blocking sync", as was discussed on discord. So sync *really* must mean blocking-the-event-loop sync not the one I had initially thought was ok.